### ISSUE TITLE
VRR in embedded breaks close to half screen frequency and bellow. fix attached

### ISSUE THREAD

**Issue by adolfotregosa** [ValveSoftware/gamescope#1513]
It seems to me that LFC (Low Framerate Compensation) or the doubling/tripling of the screen frequency isn't kicking in or close the middle of the panel frequency it breaks. I'm testing this on both an Asus Ally X and an OG Ally, and when limiting the framerate to 55 or 62 FPS, the issue becomes very noticeable during panning motion. If you toggle VRR (Variable Refresh Rate) off, the image stutters, as expected, since it's no longer synced with the screen frequency, but it's actually smoother than with VRR on. With VRR on, the image "jumps." As mentioned earlier, it seems like LFC or screen frequency doubling isn't working as expected. I can't be sure what exactly is malfunctioning because I can't directly monitor the screen's refresh rate. It could just as easily be an issue with how Gamescope is outputting frames.

I'm limiting the FPS using the Steam game launch option:
mangohud MANGOHUD_CONFIG=fps_limit=55,no_display %command%
mangohud MANGOHUD_CONFIG=fps_limit=62,no_display %command%
But there are alternatives, like:

Libstrangle: strangle 55 %command% or strangle 62 %command%
VKD3D: VKD3D_FRAME_RATE=55 %command% or VKD3D_FRAME_RATE=62 %command%
DXVK: DXVK_FRAME_RATE=55 %command% or DXVK_FRAME_RATE=62 %command%

I've included a slow-motion recording to help demonstrate the issue @55FPS: https://youtu.be/NixcItH-D14. Skip to between 1 (VRR OFF) and 3 minutes (VRR ON), and you'll clearly see the poor frame pacing at the 3-minute mark (with VRR on). I turn VRR on around minute 2:10.

The patch that fixes this issue is attached 
[revert-299bc34.zip](https://github.com/user-attachments/files/16922384/revert-299bc34.zip)

I also need to apply this patch so VRR toggle button appears for the ally and ally X.
[external_monitor.zip](https://github.com/user-attachments/files/16922396/external_monitor.zip)

**Comment by njdom24** [comment]
Is this only for DRM mode? I tested the first patch because I'm seeing this issue with all embedded backends, and it didn't improve anything.

**Comment by matte-schwartz** [comment]
Don't know the exact issue you're facing, but that patch only really helps if you're using a mangoapp overlay at the same time when you notice VRR timings going wonky.

**Comment by njdom24** [comment]
I do use mangoapp, but I'm still seeing the issue without it. I must be seeing another cause.

**Comment by matte-schwartz** [comment]
@njdom24 when you say "embedded backends", do you mean that you are launching steam within gamescope **by itself** in a TTY terminal (`gamescope -e -- steam -steamdeck`) or do you mean you are nesting gamescope within SwayWM? if it's only while nested in another WM (like Sway), it's likely you are running into https://github.com/ValveSoftware/gamescope/issues/1617 (it's not GPU exclusive).

**Comment by njdom24** [comment]
Yes, my mistake. I'm having issues with the **nested** backends (Wayland, SDL).
